finagle-core/src/main/scala/com/twitter/finagle/filter/RequestMeterFilter.scala

```diff
@@ -0,0 +1,21 @@
+package com.twitter.finagle.filter
+
+import com.twitter.concurrent.AsyncMeter
+import com.twitter.finagle.{Failure, Service, SimpleFilter}
+import com.twitter.util.{Future, Throw}
+
+/**
+  * A [[com.twitter.finagle.Filter]] that rate limits requests to a fixed rate over time by
+  * using the [[com.twitter.concurrent.AsyncMeter]] implementation. It can be used for slowing
+  * down access to throttled resources. Requests that are unable to acquire a permit are failed
+  * immediately with a [[com.twitter.finagle.Failure]] that signals a restartable or
+  * idempotent process.
+  */
+class RequestMeterFilter[Req, Rep](meter: AsyncMeter, permits: Int = 1) extends SimpleFilter[Req, Rep] {
```

Comment:
now that I think about it, how about we don't make permits configurable?  the point of having a permits argument in await is for folks who want to sometimes await on a different number of permits–since we always await on the same number, it's fine to just pick 1, imo.